python/ray/data/datasource/file_based_datasource.py

```diff
@@ -119,7 +119,7 @@ def do_write(self,
 
         _write_block_to_file = self._write_block
 
-        @ray.remote
+        @ray.remote(retry_exceptions=True)
         def write_block(write_path: str, block: Block):
```

Comment:
Can you actually make this use cached_remote_fn, and move the retry exceptions flag into the cached_remote_fn fn?